### ISSUE TITLE
[steamshovel] loop over p-frames only (Trac #1079)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1073]
The option to loop over only p-frames, which was really useful for tv mode or just showing off events, has disappeared.  Please bring it back?

Migrated from https://code.icecube.wisc.edu/ticket/1079
```json
{
    "status": "closed", 
    "changetime": "2015-08-10T20:31:17", 
    "description": "The option to loop over only p-frames, which was really useful for tv mode or just showing off events, has disappeared.  Please bring it back?", 
    "reporter": "david.schultz", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1439238677802127", 
    "component": "combo core", 
    "summary": "[steamshovel] loop over p-frames only", 
    "priority": "blocker", 
    "keywords": "", 
    "time": "2015-07-30T16:12:37", 
    "milestone": "", 
    "owner": "hdembinski", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20150731T16:15:58: hdembinski commented:

> I had removed the option, because we have filters now, which are a more general solution.
> 
> If you only want to loop over P frames, put in the filter:
>     return stream == I3Frame.Physics
> 
> Then "Loop over frames" loops only over the P frames.

**Comment by icecube-trac** [comment]
Trac update at 20150731T16:20:25: david.schultz commented:

> I think the problem with the filters is, they are powerful but complex. And users hate complexity.  Maybe have a couple of stored filters that can be chosen to handle the common tasks?
> 
> Also, the filters area looks kind of odd and just takes up space in my steamshovel view.  Is there a way to remove it from the view?

**Comment by icecube-trac** [comment]
Trac update at 20150801T12:31:28: 
* hdembinski changed _comment0 from:

> You are right, the old option was easy, but the filters are not complicated. This is an opportunity to teach people how to use them. We can make a note in the "Interface changes" on how to get the old behaviour of "Loop over P frames".
> 
> In IceCube, we are in the comfortable position that everybody knows how to do analyses in Python. Everyone should be able to write a one-liner to select their frames of interest.
> 
> If you have set up your TV display with your filter and then save the session, the filter is saved with it. So really, one only has to set up the system once.
> 
> How to remove the filter view: you can just drag the separator handle between the "settings view" and the "filter view" down, at some point it will snap to the baseline and the "filter view" together with the play/stop/... buttons is hidden.

to:

> 1438432310722368


* hdembinski commented:

> You are right, the old option was easy, but the filters are not complicated. This is an opportunity to teach people how to use them. We can make a note in the "Interface changes" on how to get the old behaviour of "Loop over P frames".
> 
> In IceCube, we are in the comfortable position that everybody knows how to do analyses in Python. Everyone should be able to write a one-liner to select their frames of interest.
> 
> If you have set up your TV display with your filter and then save the session, the filter is saved with it. So really, one only has to set up the system once.
> 
> How to hide the filter view: you can just drag the separator handle between the "settings view" and the "filter view" down, at some point it will snap to the baseline and the "filter view" together with the play/stop/... buttons is hidden.

**Comment by icecube-trac** [comment]
Trac update at 20150801T13:50:40: david.schultz commented:

> I would argue that not everyone knows or is comfortable in Python. Examples:
> 
> * outreach
> * undergrads and other "interns"
> * several PIs

**Comment by icecube-trac** [comment]
Trac update at 20150810T19:54:01: 
* hdembinski changed status from "new" to "closed"
* hdembinski changed resolution from "" to "fixed"
* hdembinski commented:

> In changeset 136040:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="136040"
> loop-over-X-frames button is back, closes #1073
> ```

**Comment by icecube-trac** [comment]
Trac update at 20150810T20:31:17: 
* olivas commented:

> Milestone offline V15-07-00 deleted


* olivas changed milestone from "offline V15-07-00" to ""
